### PR TITLE
Cuda TF import fixes

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?

Fix TF import data loading for float16/bfloat16
Remove commented code
Fix non functrace build issue with CPU

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.konduit.ai/multi-project/how-to-guides/contribute/eclipse-contributors) page for details
- [X ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ X] Created tests for any significant new code additions.
- [X ] Relevant tests for your changes are passing.

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/memory/deallocation/DeallocatorService.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ public long nextValue() {
      */
     public long pickObject(@NonNull Deallocatable deallocatable) {
         if(noPointerGc) {
-            log.trace("Deallocation turned off. Reference " + deallocatable.getUniqueId() + " will need to be de allocated manually.");
+            //log.trace("Deallocation turned off. Reference " + deallocatable.getUniqueId() + " will need to be de allocated manually.");
```
Comment (treo) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/memory/deallocation/DeallocatorService.java]: is this left commented out intentionally?

File: nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda-preset/src/main/java/org/nd4j/presets/cuda/Nd4jCudaPresets.java [hunk 2]
```diff
@@ -152,7 +152,8 @@ public void init(Logger logger, java.util.Properties properties, String encoding
         String platform = properties.getProperty("platform");
         List<String> preloads = properties.get("platform.preload");
         List<String> resources = properties.get("platform.preloadresource");
-
+        boolean funcTrace = System.getProperty("libnd4j.calltrace","OFF").equalsIgnoreCase("ON");
+        System.out.println("Functrace on: " + funcTrace);
```
Comment (treo) [inline nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda-preset/src/main/java/org/nd4j/presets/cuda/Nd4jCudaPresets.java:156]: I guess this print is a left over from debugging?

File: nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/linalg/jcublas/ops/executioner/CudaExecutioner.java [hunk 3]
```diff
@@ -2137,7 +2138,7 @@ public DataBuffer createShapeInfo(long[] shape, long[] stride, long elementWiseS
 
         val result = new CudaLongDataBuffer(nativeOps.getConstantShapeBufferPrimary(dbf), nativeOps.getConstantShapeBufferSpecial(dbf), Shape.shapeInfoLength(shape.length));
 
-        nativeOps.deleteConstantShapeBuffer(dbf);
+        //nativeOps.deleteConstantShapeBuffer(dbf);
```
Comment (treo) [inline nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/linalg/jcublas/ops/executioner/CudaExecutioner.java]: is this left commented out intentionally?

**Review by treo — CHANGES_REQUESTED** [review]

File: nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda-preset/src/main/java/org/nd4j/presets/cuda/Nd4jCudaPresets.java [hunk 4]
```diff
@@ -182,6 +183,7 @@ public void init(Logger logger, java.util.Properties properties, String encoding
     public void map(InfoMap infoMap) {
         //whether to include the SD_GCC_FUNCTRACE definition in the build. Not needed if we're not enabling the profiler.
         boolean funcTrace = System.getProperty("libnd4j.calltrace","OFF").equalsIgnoreCase("ON");
+        System.out.println("Functrace on: " + funcTrace);
```
Comment (treo) [inline nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda-preset/src/main/java/org/nd4j/presets/cuda/Nd4jCudaPresets.java:186]: I guess this print is a left over from debugging?

Comment (agibsonccc) [reply nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda-preset/src/main/java/org/nd4j/presets/cuda/Nd4jCudaPresets.java:186]: No I'm leaving that in to verify whether the presets detect functrace is on. The rest I've cleaned up.

**Review by treo — APPROVED** [review]

**Pull request merged**